packages/node/src/executors/execute/execute.impl.ts

```diff
@@ -32,7 +32,30 @@ export interface NodeExecuteBuilderOptions {
 }
 
 let subProcess: ChildProcess = null;
-
+/**
+ * Ensure subprocess is cleaned up when parent process exits as a result of
+ * process.exit() rather than from a signal (like from <ctrl-c>).
+ *
+ * When the process is killed by a signal, that signal is forwarded to any
+ * forked processes ensuring the children are also shut down. But with
+ * process.exit() there is no signal to forward so the forked process is
+ * left running.
+ *
+ * This currently happens in our testcafe builder where we start the server
+ * using the execute builder and then run our tests against that server. At
+ * the end of the test run, we complete the builder observable using `take(1)`,
+ * resulting in the Nx cli calling `process.exit(builderOutput.succes ? 0 : 1)`
+ * orphaning our server process.
+ *
+ * NOTE: We can't use treeKill(subProcess.pid, 'SIGTERM') because it performs
+ *       async operations which are disallowed in  the `exit` event.
+ */
+process.on('exit', () => {
```

Comment:
This should be handled here right?

https://github.com/nrwl/nx/pull/6733/files#diff-8b1daad050dbc745646b89f91cd20c497a23f51caf7eeaddb96a517a80b8f01fL45

It's curious that it's calling `process.exit` and not `subProcess.exit` :thinking: 

@AgentEnder @nartc can you take a look please?